libr/util/print_code.c

```diff
@@ -0,0 +1,281 @@
+/* radare - LGPL - Copyright 2007-2020 - pancake */
+
+#include <r_util.h>
+#include <r_util/r_print.h>
+
+static const char* bits_to_c_code_fmtstr(int bits) {
+	switch (bits) {
+	case 16:
+		return "0x%04x";
+	case 32:
+		return "0x%08xU";
+	case 64:
+		return "0x%016" PFMT64x "ULL";
+	}
+	return "0x%02x";
+}
+
+static void print_c_instructions(RPrint *p, ut64 addr, const ut8 *buf, int len) {
```

Comment:
please see https://github.com/radareorg/radare2/pull/16792#pullrequestreview-408955758, the implementation of this function is (in my opinion) very hard to understand.

Reply:
To do what you want we need more apis accessible from rprint. Thats the only way we can do that here. I csn refactor the code to look better if u want. But its pretty clean as it is right now. We have zillions of things to refactor and cleanup everywhere in r2land. So this pr can become and endless monster of unrelated changes just to add a command

Reply:
I know, but this is new code so it has to be good. No reason to introduce code that is already hard and already needs a lot of refactoring.

I think something like https://pastebin.com/6seAgYV6 looks much easier to read.

Reply:
Fantastic suggestion @ret2libc. The code is very clean and clear